src/util/json.h

```diff
@@ -27,13 +30,69 @@ namespace nlohmann {
         }
 
         static void from_json(const nlohmann::json& j, std::optional<T>& opt) {
-          if (j.is_null() || !j["has_value"].get<bool>()){
+          if (j.is_null()){
             opt = std::nullopt;
           } else {
-            opt = j["value"].get<T>();
+            opt = j.get<T>();
           }
         }
     };
 }
 
+// Added support for serializing `std::monostate` using `nlohmann::json`.
+namespace nlohmann {
+    template<>
+    struct adl_serializer<std::monostate>{
+        static void to_json(nlohmann::json& j,
+            const std::monostate&) {
+          // Monostate is just an empty placeholder value.
+          j = nullptr;
+        }
+
+        static void from_json(const nlohmann::json&,
+            std::monostate&) {
+          // Monostate holds no values, so the given monostate is already
+          // identical to the serialized one.
+          // TODO Add an exception here, if the j doesn't contain a nullptr.
+          // I mean, there is literally no sense in trying to convert any actual
+          // value to a std::monostate, so there must be a bug, if somebody tries
+          // to!
+        }
+    };
+}
+
+// Added support for serializing `std::variant` using `nlohmann::json`.
+namespace nlohmann {
+    template <typename... Types>
+    struct adl_serializer<std::optional<Types...>> {
+        static void to_json(nlohmann::json& j,
+            const std::variant<Types...>& var) {
```

Comment:
Doesn't this have to be `adl_serializer<std::variant<....>>` (classic copy and paste bug?)

Reply:
You're completely right. I fixed it and will commit it, once I cleaned up the code and tested the new feature more thoroughly.

Reply:
Done, see [7019669](https://github.com/ad-freiburg/qlever/pull/860/commits/70196692c25671279c4bacfbd407aaab4600e015).